### DISCUSSION TITLE
can we use this in production builds?

### DISCUSSION THREAD

**Discussion by aroman** [jotaijs/jotai-devtools#102] [General]
is there a reason this tool can't work in production builds?

**Comment by arjunvegda** [comment]
Hi @aroman,

Apologies for the delay! This is because the methods we rely on to power the dev tools in the store are available in dev env. only. This is to keep Jotai's bundle size and memory consumption low in production env. 

What is your use case?

**Reply by valeriobelli** [reply]
Hello all 👋🏽 
I can give my opinion. I find it useful for debugging in a pre-production environment (e.g., letting the QA department run their investigations by looking at the internal app state). Theoretically, by keeping almost the current package interface, declaring `./internal__devtools` in the `exports` and removing this [additional](https://github.com/jotaijs/jotai-devtools/blob/main/src/DevTools/DevTools.tsx#L130-L136) check on the env should permit this. 🙂 

Other players also allow this as an advanced use case. This is an [example](https://tanstack.com/query/v5/docs/framework/react/devtools#devtools-in-production) of how TanStack Query solved this issue.

**Reply by ywongweb** [reply]
I would also like to see Jotai DevTools support debugging in production.

Production issues are often unplanned, high-priority, and time-sensitive. Without DevTools, engineers lose access to a critical tool that they frequently rely on during their regular development debugging process. This absence means they have to switch to other less used methods thus increasing mental switching costs and adding more variables to an already high-pressured situation.

imho being able to debug in production can even be a make-or-break feature for projects with advanced state needs.

**Comment by dai-shi** [comment]
I may consider adding a capability to create dev store without depending on envvars. It's not very easy, but something in my head.
